### PR TITLE
Add appservice tag for dedicated .NET 8 in proc image

### PR DESCRIPTION
**Pull request by aishwaryabh**
This PR adds an appservice tag to the `publish.yml` and `republish.yml` files so that the tooling feed can consume these changes.

### PR information

- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

File: host/4/publish.yml [hunk 1]
```diff
@@ -80,8 +80,13 @@ steps:
   - bash: |
       set -e
       docker pull $SOURCE_REGISTRY/dotnet:$(PrivateVersion)-dotnet8
+      docker pull $SOURCE_REGISTRY/dotnet:$(PrivateVersion)-dotnet8-appservice
```
Comment (pragnagopa) [inline host/4/publish.yml]: @Cooperlink should this be 8.0?

Comment (CooperLink) [reply host/4/publish.yml]: I do not know it depends on what the platform is pulling for this stack. That is a good callout. I can check the code/ look for logs/ find an old linux dedicated stamp that is kicking around maybe

Comment (CooperLink) [reply host/4/publish.yml]: This should be 8.0 to match the expected tag in Antares. Thank you Pragna

Comment (mattchenderson) [reply host/4/publish.yml]: Per discussion on Teams, it seems there would not be a difference in behavior from that, and other work will be needed to account for whatever form the tag takes. Therefore, the current form here should be equally fine.

Comment (aishwaryabh) [reply host/4/publish.yml]: I went ahead and updated all the tags to end with 8.0 (including the existing images)

**Review by pragnagopa — CHANGES_REQUESTED** [review]
Lets wait for ANT changes to support .Net 8 inproc

**Comment by pragnagopa** [comment]
addressed in https://github.com/Azure/azure-functions-docker/pull/1096 ; closing this PR

@Fabiocav @CooperLink  @aishwaryabh  @mattchenderson

**Pull request closed without merging**